packages/connector-tencent-email/src/index.ts

```diff
@@ -0,0 +1,148 @@
+import type {
+  CreateConnector,
+  EmailConnector,
+  GetConnectorConfig,
+  SendMessageFunction,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorType,
+  validateConfig,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { isErrorResponse, request } from '@/http';
+import type { MailTemplateType, SendTencentMailConfig, ValueNames } from '@/types';
+import { sendTencentMailConfigGuard } from '@/types';
+
+import { defaultMetadata } from './constant';
+
+function getTemplate(templates: MailTemplateType[], usageType: string): MailTemplateType {
+  const template = templates.find((template) => template.usageType === usageType);
+  assert(
+    template,
+    new ConnectorError(
+      ConnectorErrorCodes.TemplateNotFound,
+      `Template not found for type: ${usageType}`
+    )
+  );
+
+  return template;
+}
+
+function buildFromName(fromName: string, fromAddress: string): string {
+  return fromName.length > 0 ? `${fromName} <${fromAddress}>` : fromAddress;
+}
+
+function buildTemplatePayload(
+  template: MailTemplateType,
+  templateParameters: Record<typeof ValueNames[number], string>
+): Record<string, string> {
+  return template.params.reduce<Record<string, string>>((accumulator, parameter) => {
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    accumulator[parameter.name] = templateParameters[parameter.value];
+
+    return accumulator;
+  }, {});
+}
+
+function transformString(...values: Array<string | undefined>): string {
+  return values.find((item) => item && item.length > 0) ?? '';
+}
+
+const sendMessage =
+  (getConfig: GetConnectorConfig): SendMessageFunction =>
+  async (data, inputConfig) => {
+    const { to, type, payload } = data;
+    const config = inputConfig ?? (await getConfig(defaultMetadata.id));
+    validateConfig<SendTencentMailConfig>(config, sendTencentMailConfigGuard);
+    const { accessKeyId, accessKeySecret, region, fromAddress, fromName, replyAddress, templates } =
+      config;
+
+    const mailTemplate: MailTemplateType = getTemplate(templates, type);
+
+    const fromAddressValue = transformString(mailTemplate.from?.fromAddress, fromAddress);
+
+    assert(
+      fromAddressValue.length > 0,
+      new ConnectorError(
+        ConnectorErrorCodes.NotImplemented,
```

Comment:
Is there a better error code than `NotImplemented`?

Reply:
Yeah, I see the logic. But what I meant was the error code. When the `from` is not specified, it should have been something else rather than `Not implemented`, IMO